src/Account.sol

```diff
@@ -798,9 +806,13 @@ contract Account is IAccount, OpsReady, Auth, Initializable {
         ConditionalOrder memory conditionalOrder =
             getConditionalOrder(_conditionalOrderId);
 
-        // delete conditional order from conditional orders
+        // remove conditional order from internal accounting
         delete conditionalOrders[_conditionalOrderId];
 
+        // remove gelato task from their accounting
+        /// @dev will revert if task id does not exist {Automate.cancelTask: Task not found}
+        IOps(OPS).cancelTask({taskId: conditionalOrder.gelatoTaskId});
```

Comment:
Unless I missed it can you add a test for this?

Reply:
I added it here: https://github.com/Kwenta/margin-manager/blob/061dd7f5101ae9c5dc09d5f3ed948ce966fd1e29/test/integration/order.behavior.t.sol#L744 and furthermore, added a test when execution fails ensuring the task remains here: https://github.com/Kwenta/margin-manager/blob/061dd7f5101ae9c5dc09d5f3ed948ce966fd1e29/test/integration/order.behavior.t.sol#L786